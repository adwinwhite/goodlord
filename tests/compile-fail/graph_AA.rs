use deadlock_proof::LockAfter;
use deadlock_proof::impl_lock_order;

struct A;

impl_lock_order!(A => A);

fn main() {}
