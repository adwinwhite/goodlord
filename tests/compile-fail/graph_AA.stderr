error[E0119]: conflicting implementations of trait `LockAfter<A>` for type `A`
 --> tests/compile-fail/graph_AA.rs:6:1
  |
6 | impl_lock_order!(A => A);
  | ^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | first implementation here
  | conflicting implementation for `A`
  |
  = note: this error originates in the macro `impl_lock_order` (in Nightly builds, run with -Z macro-backtrace for more info)
